source/op/descrpt_se_a_mask.cc

```diff
@@ -0,0 +1,337 @@
+                        // derive of the component z/r2
+                        descrpt_deriv_atom[jj * 12 + 9] = 0.;  // on x.
+                        descrpt_deriv_atom[jj * 12 + 10] = 0.; // on y.
+                        descrpt_deriv_atom[jj * 12 + 11] = 0.; // on z.
+                        rij_atom[jj * 3 + 0] = 0.;
+                        rij_atom[jj * 3 + 1] = 0.;
+                        rij_atom[jj * 3 + 2] = 0.;
+                        continue;
+                    }
+
+                    for (int dd = 0; dd < 3; dd++)
+                    {
+                        temp_diff[dd] = d_coord3[j_idx * 3 + dd] - rloc[dd];
+                        rij_atom[jj * 3 + dd] = temp_diff[dd];
+                    }
+
+                    temp_rr = deepmd::dot3<compute_t>(temp_diff, temp_diff);
+
+                    compute_t x = temp_diff[0];
+                    compute_t y = temp_diff[1];
+                    compute_t z = temp_diff[2];
+
+                    // r^2
+                    compute_t nr2 = temp_rr;
+                    // 1/r
+                    compute_t inr = 1. / sqrt(nr2);
+                    // r
+                    compute_t nr = nr2 * inr;
+                    // 1/r^2
+                    compute_t inr2 = inr * inr;
+                    // 1/r^4
+                    compute_t inr4 = inr2 * inr2;
+                    // 1/r^3
+                    compute_t inr3 = inr * inr2;
+                    // 1./rr, cos(theta), cos(phi), sin(phi)
+                    descrpt_atom[jj * 4 + 0] = 1. / nr;
+                    descrpt_atom[jj * 4 + 1] = x / nr2;
+                    descrpt_atom[jj * 4 + 2] = y / nr2;
+                    descrpt_atom[jj * 4 + 3] = z / nr2;
+                    // derive of the component 1/r
+                    descrpt_deriv_atom[jj * 12 + 0] = x * inr3;
+                    descrpt_deriv_atom[jj * 12 + 1] = y * inr3;
+                    descrpt_deriv_atom[jj * 12 + 2] = z * inr3;
+                    // derive of the component x/r2
+                    descrpt_deriv_atom[jj * 12 + 3] = 2. * x * x * inr4 - inr2; // on x.
+                    descrpt_deriv_atom[jj * 12 + 4] = 2. * x * y * inr4;        // on y.
+                    descrpt_deriv_atom[jj * 12 + 5] = 2. * x * z * inr4;        // on z.
+                    // derive of the component y/r2
+                    descrpt_deriv_atom[jj * 12 + 6] = 2. * y * x * inr4;        // on x.
+                    descrpt_deriv_atom[jj * 12 + 7] = 2. * y * y * inr4 - inr2; // on y.
+                    descrpt_deriv_atom[jj * 12 + 8] = 2. * y * z * inr4;        // on z.
+                    // derive of the component z/r2
+                    descrpt_deriv_atom[jj * 12 + 9] = 2. * z * x * inr4;         // on x.
+                    descrpt_deriv_atom[jj * 12 + 10] = 2. * z * y * inr4;        // on y.
+                    descrpt_deriv_atom[jj * 12 + 11] = 2. * z * z * inr4 - inr2; // on z.
```

Comment:
`1.` and `2.` is `float64` but it should be `compute_t` before it multiplies by another `compute_t`.